### ISSUE TITLE
cmd/protoc-gen-go: tests run very slowly under race detector

### ISSUE THREAD

**Issue by dsnet** [golang/protobuf#717]
(low priority; helpful for developer experience)

On Go1.11, run the following on (ded92f8033e32020261f3e99788af9641c360239):
```
$ go test google.golang.org/proto/cmd/protoc-gen-go -count=1
ok  	google.golang.org/proto/cmd/protoc-gen-go	0.249s
$ go test google.golang.org/proto/cmd/protoc-gen-go -count=1 -race
ok  	google.golang.org/proto/cmd/protoc-gen-go	17.641s
```

The test doesn't seems to require alot of concurrency, so it's odd that it takes 17.6 seconds with the race detector enabled. We should investigate why.

\cc @neild

**Comment by dsnet** [comment]
The cause is because compiling under the race detector inserts a 1 second sleep at the end of the binary:
```go
$ cat main.go
package main
func main() {}
$ go build -race main.go
$ time ./main
real	0m1.018s
user	0m0.007s
sys	0m0.007s
$ strace ./main
...
nanosleep({1, 0}, NULL) = 0
``` 

Part of the operation of the protoc-gen-go GoldenTest is that it passes itself to `protoc` as if it were the `protoc-gen-go` binary. However, the `nanosleep` syscall at the end of the `protoc-gen-go.test` binary means that every invocation will take at least 1 second.

Things we can consider doing:
* Not running golden tests under the race detector
* Run all the golden tests in parallel, since a a sleep implies that we are neither CPU, nor IO bound.

**Comment by neild** [comment]
I think just not running them under the race detector is fine. The generator isn't concurrent, so there isn't anything interesting for it to find there.

**Comment by dsnet** [comment]
Fixed in http://golang.org/cl/137037.
